### ISSUE TITLE
Agent forwarding broken on Ubuntu

### ISSUE THREAD

**Issue by garethblower** [microsoft/vscode#177193]
Type: <b>Bug</b>

Start a remote session on an Ubuntu 22.04 machine from an Ubuntu 22.04 machine. Go to the terminal window, and type `ls -l ${SSH_AUTH_SOCK}`. Observe that this is a valid symlink. Open another remote VSC window, or shutdown and restart the current remote session. Now repeat the above, and note that the output of `ls -l ${SSH_AUTH_SOCK}` is `ls: cannot access '/run/user/1001/vscode-ssh-auth-sock-747671732': No such file or directory`. GitHub operations from VSC will now no longer work.

This has been broken for the last few versions of VSC.

VS Code version: Code 1.76.1 (5e805b79fcb6ba4c2d23712967df89a089da575b, 2023-03-08T16:31:31.010Z)
OS version: Linux x64 5.15.0-67-generic
Modes:
Sandboxed: No
Remote OS version: Linux x64 5.15.0-67-generic
Remote OS version: Linux x64 5.15.0-67-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 x 2200)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 2, 1|
|Memory (System)|46.82GB (28.72GB free)|
|Process Argv|--crash-reporter-id d416c91c-34b5-4fa6-96be-a25c1236c719|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|wayland|

|Item|Value|
|---|---|
|Remote|SSH: dev|
|OS|Linux x64 5.15.0-67-generic|
|CPUs|Intel Xeon Processor (Skylake, IBRS) (10 x 2099)|
|Memory (System)|47.05GB (42.22GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: dev|
|OS|Linux x64 5.15.0-67-generic|
|CPUs|Intel Xeon Processor (Skylake, IBRS) (10 x 2099)|
|Memory (System)|47.05GB (42.22GB free)|
|VM|0%|
</details><details><summary>Extensions (60)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.2
Bookmarks|ale|13.3.1
vsc-material-theme|Equ|33.6.0
vsc-material-theme-icons|equ|2.6.1
bats|jet|0.1.9
vscode-todo-highlight|jgc|2.0.4
Kotlin|mat|1.7.1
jupyter-keymap|ms-|1.1.0
remote-containers|ms-|0.282.0
remote-ssh|ms-|0.98.0
remote-ssh-edit|ms-|0.84.0
remote-explorer|ms-|0.2.0
shellman|Rem|5.6.0
shebang-snippets|rpi|0.1.4
rewrap|stk|1.16.3
better-comments|aar|3.0.2
Bookmarks|ale|13.3.1
github-markdown-preview|bie|0.3.0
markdown-checkbox|bie|0.4.0
markdown-emoji|bie|0.3.0
markdown-footnotes|bie|0.1.1
markdown-mermaid|bie|1.17.7
markdown-preview-github-styles|bie|1.0.1
markdown-yaml-preamble|bie|0.1.0
gitlens|eam|13.3.2
copilot|Git|1.77.9225
gitlab-workflow|Git|3.60.0
go|gol|0.38.0
cloudcode|goo|1.21.3
todo-tree|Gru|0.0.224
bats|jet|0.1.9
bash-ide-vscode|mad|1.35.0
Kotlin|mat|1.7.1
rainbow-csv|mec|3.5.0
json|Mee|0.1.2
vscode-docker|ms-|1.24.0
vscode-kubernetes-tools|ms-|1.3.11
isort|ms-|2022.8.0
python|ms-|2023.4.1
vscode-pylance|ms-|2023.3.20
jupyter|ms-|2023.2.1200692131
jupyter-keymap|ms-|1.1.0
jupyter-renderers|ms-|1.0.15
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
cpptools|ms-|1.14.4
makefile-tools|ms-|0.6.0
vsliveshare|ms-|1.0.5834
biosyntax|rea|0.0.11
vscode-yaml|red|1.12.1
shellman|Rem|5.6.0
bash-debug|rog|0.3.9
shebang-snippets|rpi|0.1.4
rust-analyzer|rus|0.3.1435
jinjahtml|sam|0.20.0
crates|ser|0.5.10
markdown-preview-enhanced|shd|0.6.8
vscode-lldb|vad|1.9.0
docker|vsc|1.0.0
yaml|vsc|1.0.0

(4 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewvextcfv2:30681851
azure-dev_surveyone:30548225
pyindex848:30662994
nodejswelcome1:30587005
282f8724:30602487
pyind779:30671433
f6dab269:30613381
vscrp:30673768
pythonsymbol12:30671437
vsccsb:30677849
vscodedisable:30660115
pythonb192cf:30669361
funwalk2:30682974
pythonms35:30686772

```

</details>

**Comment by madalinignisca** [comment]
@eleanorjboyd, #168202 is not the same like this one.

I think there are other tickets with SSH Agent not working on Ubuntu 22.04 remote host. Maybe people from Canonical know what they changed, as towards Debian 12 works, but not Ubuntu 22.04.

Ubuntu 22.04 was started from older testing branch of Debian 12 (not yet released stable), but any Ubuntu 22.04 remote server, or any image for Devcontainers based on it, ain't working.

I ditched all work with Ubuntu considering that I might face other critical problems in the future and moved all work on Debian, with it's philosophy of hard to accept changes, and if changes happen you really know about it.

**Comment by eleanorjboyd** [comment]
Hello! To clarify did you switch to Debian and now the problem doesn't exist or does the problem still exist following your switch? If you are still seeing the problem the could you please send your remote-ssh logs? You can find the logs by going to `view` in the menu bar, then selecting `output`, and finally, once the bottom window pops up, select  `remote-ssh` from the dropdown on the right. Thanks!
